Project/Assets/Prairie/Framework/Script/Twine/TwineNode.cs

```diff
@@ -52,152 +55,182 @@ public class TwineNode : MonoBehaviour
+                Rect frame = new Rect (10, 10+index*100, frameWidth, frameHeight);
+
+                GUIStyle style = new GUIStyle (GUI.skin.box);
+                style.wordWrap = true;
+                style.fixedWidth = frameWidth;
+
+                GUI.BeginGroup(new Rect(20, 10+index*frameHeight, frameWidth + 10, frameHeight));
+
+                scrollPosition = GUILayout.BeginScrollView(scrollPosition, GUILayout.Width(frameWidth + 10), GUILayout.Height(frameHeight));
+                content = content.TrimEnd();
+                GUILayout.Label(new GUIContent(this.content), style, GUILayout.Width(frameWidth - 50), GUILayout.ExpandHeight(true));
+                if (isDecisionNode) {
+                    GUIStyle decisionHintStyle = new GUIStyle (style);
+                    decisionHintStyle.fontStyle = FontStyle.BoldAndItalic;
+
+                    if (!isOptionsGuiOpen) {
+                        GUILayout.Box ("Press TAB to progress in the story...", decisionHintStyle);
+                    } else {
+                        GUILayout.Box ("Press TAB to scroll, E to close, ENTER to choose", decisionHintStyle);
+                    }
+                }
 
-            GUI.EndGroup();
+                if (this.isOptionsGuiOpen) {
+                    // Draw list of buttons for the possible children nodes to visit:
+                    GUIStyle optionButtonStyle = new GUIStyle (GUI.skin.button);
+                    optionButtonStyle.fontStyle = FontStyle.Italic;
+                    optionButtonStyle.wordWrap = true;
 
-        }
-        else if (this.enabled && this.isMinimized) // Don't draw any boxes, just a tiny icon
-        {
+                    // Set highlighted button to have green text (this state is called `onNormal`):
+                    optionButtonStyle.onNormal.textColor = Color.white;
+                    // Set non-highlighted buttons to have grayed out text (state is called `normal`)
+                    optionButtonStyle.normal.textColor = Color.gray;
 
-            // Draw minimized GUI instead
-            Rect frame = new Rect(10, 10, 10, 10);
+                    selectedOptionIndex = GUILayout.SelectionGrid(selectedOptionIndex, this.childrenNames, 1, optionButtonStyle);
+                }
+                GUILayout.EndScrollView();
+                GUI.EndGroup ();
 
-            GUI.Box(frame, "");
+            }
+            else if (this.enabled && !this.isMinimized && !allMinimized) {
+                float frameWidth = Math.Min(Screen.width / 3, 350);
+                float frameHeight = Math.Min(Screen.height / 2, 500);
+                Rect frame = new Rect (10, 10, frameWidth, frameHeight);
+                GUI.BeginGroup (frame);
+                GUIStyle style = new GUIStyle (GUI.skin.box);
+                style.wordWrap = true;
+                style.fixedWidth = frameWidth;
+                GUILayout.Box (this.content, style);
+
+                if (isDecisionNode) {
```

Comment:
Seems like you are doing the exact same thing in both if (above) and else if here. Please put them outside of the if statement or compile them into a function instead of copying a large chunk of code in close by places.